jte/src/main/java/gg/jte/resolve/CompositeCodeResolver.java

```diff
@@ -0,0 +1,68 @@
+package gg.jte.resolve;
+
+
+import gg.jte.CodeResolver;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Resolves template code from multiple other resolvers
+ */
+public class CompositeCodeResolver implements CodeResolver {
+
+    private final List<CodeResolver> codeResolvers;
+
+    public CompositeCodeResolver(List<CodeResolver> codeResolvers) {
```

Comment:
For ease of use I would add an overloaded constructor that takes a varargs parameter. i.e.

```java
public CompositeCodeResolver(CodeResolver... codeResolvers) {
    this(List.of(codeResolvers));
}
```